stuffer/s2n_stuffer.c

```diff
@@ -52,7 +52,14 @@ bool s2n_stuffer_reservation_is_valid(const struct s2n_stuffer_reservation* rese
         return false;
     }
     const struct s2n_stuffer stuffer_obj = *(reserve_obj.stuffer);
-    return S2N_MEM_IS_WRITABLE(stuffer_obj.blob.data + reserve_obj.write_cursor, reserve_obj.length);
+    if (stuffer_obj.blob.size < reserve_obj.length) {
+        return false;
+    }
+    return S2N_IMPLIES(
+        reserve_obj.length > 0,
+        (reserve_obj.write_cursor < stuffer_obj.blob.size &&
+         S2N_MEM_IS_WRITABLE(stuffer_obj.blob.data + reserve_obj.write_cursor, reserve_obj.length))
+    );
```

Comment:
@danielsn @camshaft LGTM, but do you also agree with these changes?

Reply:
With the additional change per CR by @feliperodri and myself, LGTM too.

Reply:
When would it ever make sense for a reservation to be `length = 0`? I would expect to return an error in that case.